src/vs/workbench/api/browser/mainThreadChatAgents2.ts

```diff
@@ -245,6 +245,11 @@ export class MainThreadChatAgents2 extends Disposable implements MainThreadChatA
 		chunks.forEach(item => {
 			const [progress, responsePartHandle] = Array.isArray(item) ? item : [item];
 
+			if (progress.kind === 'clearToPreviousToolInvocation') {
+				chatProgressParts.push({ kind: 'clearToPreviousToolInvocation', reason: progress.reason });
```

Comment:
Was this block necessary? This one can probably be handled by the general case right?